futures-util/src/async_await/join.rs

```diff
@@ -1,5 +1,25 @@
-//! The `join` macro.
 
+/// The `join` macro.
+/// 
+/// This macro polls multiple futures simultaneously, returning a tuple
+/// of all results once complete. While `join!(a, b)` is similar to `(await!(a), await!(b))`,
+/// unlike the latter `join!` can make progress on multiple pending futures at once.
```

Comment:
- "~~This macro~~ Polls multiple futures..." Rustdoc shows that it's a macro right above the text. Also, it always looks a bit odd if we have multiple descriptions next to each other and every one of them starts with "This marcro/method/function ..."
- Make a new paragraph after the first sentence. The first sentence is a great summary sentence